### DISCUSSION TITLE
`no-unsafe-argument `, https://typescript-eslint.io/rules/no-unsafe-argument/

### DISCUSSION THREAD

**Discussion by jpike88** [biomejs/biome#586] [Rule suggestion]
It would be good as a gradual tightening of types, to have a rule that bans the passing as an item with type any or any[] as a parameter to another function, even if the function has a parameter with type `any` in its definition.

This allows for the flexibility of having such functions remain, but ensures that stuff going in was coerced to an explicit type beforehand.

**Comment by unvalley** [comment]
Hi, isn't [noExplictAny](https://biomejs.dev/linter/rules/no-explicit-any/) enough? I would appreciate it if you could give me an example.

**Reply by aselcuktuncer** [reply]
hey, it's been a while, but I'd like to leave a comment. i have a function like this:
```
const getErrorMessage = () => {
    try {
      const parsedErrorMessage = JSON.parse(errorMessage ?? '');
      if (Array.isArray(parsedErrorMessage)) {
        return parsedErrorMessage[0]?.message; // here the return type is 'any'
      }
      return errorMessage ?? '';
    } catch (e) {
      console.error(e);
      return errorMessage ?? '';
    }
  };
```
and later i used this function here and it did not throw any linting error. but here I was expecting an error like no-any-as-parameter and to be forced to specify typeof 'parsedErrorMessage[0]?.message'
`// message?: string | undefined`
`<ErrorDisplay message={getErrorMessage()} />`

btw i am using the recommended lint rules.

**Reply by jpike88** [reply]
errorMessage ?? ''


That makes me sad. No comment otherwise

**Reply by douglasg14b** [reply]
No, `any` can sneak in from other sources that are not an explicit declaration.

**Comment by jpike88** [comment]
there's a relvant eslint rule for this now:
 https://typescript-eslint.io/rules/no-unsafe-argument/
